src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -794,6 +832,8 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		if (this._index !== newIndex) {
 			this._index = newIndex;
 			this._onDidGroupChange.fire({ kind: GroupChangeKind.GROUP_INDEX });
+			// TODO @lramos15 ENRICH THE MODEL TO LEARN ABOUT INDEX CHANGES THIS IS A HACK
```

Comment:
I think this is similar to the locked state which applies to the entire group and not a tab within.

Reply:
I'm not sure I'm following. The model is not sure of when the group index is moved.